src/Vizkit3dWorld.cpp

```diff
@@ -18,27 +18,37 @@
 
 namespace vizkit3d_world {
 
-Vizkit3dWorld::Vizkit3dWorld(std::string path, 
+Vizkit3dWorld::Vizkit3dWorld() {
+
+}
+
+Vizkit3dWorld::~Vizkit3dWorld()
+{
+}
+
+void Vizkit3dWorld::initialize(std::string path,
```

Comment:
I'm failing to see the purpose of initialize/deInitialize. Can't you have the same change in logic you currently have, but keeping the code in constructor/destructor ?

Reply:
I was working with @romulogcerqueira.  We don't want to instantiate and delete the object for each configure/cleanup running states. Also, as soon as possible, we are going to replace the variable `Vizkit3dWorld *vizkit3dWorld` from pointer to a static instance (`Vizkit3dWorld vizkit3dWorld`).

Reply:
You essentially moved the expensive parts of construction and destruction in separate methods, which makes new/delete super-cheap and initialize/cleanup super-expensive.

I.e. apart from duplicating the natural lifecycle of a C++ object (and thus making the code that much harder to understand), I really don't see any benefit in what you've done.

Reply:
> You essentially moved the expensive parts of construction and destruction in separate methods, which makes new/delete super-cheap and initialize/cleanup super-expensive.

I was just thinking a way to avoid the allocation in configureHook and replace the dynamic allocation of Vizkit3dWorld by static allocation later. But I'll follow your suggestion, we are going to keep the stuff in constructor / destructor and remove the initialize / deinitialize method.